synapse_auto_accept_invite/__init__.py

```diff
@@ -13,19 +13,39 @@
 # limitations under the License.
 from typing import Any, Dict
 
+import attr
 from synapse.module_api import EventBase, ModuleApi
 
 
+@attr.s(auto_attribs=True, frozen=True)
+class InviteAutoAccepterConfig:
+    accept_invites_only_for_direct_messages: bool = False
+
+
 class InviteAutoAccepter:
-    def __init__(self, config: Dict[str, Any], api: ModuleApi):
+    def __init__(self, config: InviteAutoAccepterConfig, api: ModuleApi):
         # Keep a reference to the Module API.
         self._api = api
+        self._config = config
 
         # Register the callback.
         self._api.register_third_party_rules_callbacks(
             on_new_event=self.on_new_event,
         )
 
+    @staticmethod
+    def parse_config(config: Dict[str, Any]) -> InviteAutoAccepterConfig:
+        """Checks that the required fields are present and at a correct value, and
+        instantiates a InviteAutoAccepterConfig.
+        Args:
+            config: The raw configuration dict.
+        Returns:
+            A InviteAutoAccepterConfig generated from this configuration
+        """
+        return InviteAutoAccepterConfig(
+            config.get("accept_invites_only_for_direct_messages", False)
+        )
```

Comment:
Just to slightly clarify this bit - might not make a big difference but helpful if we need to add more config later.

```suggestion
        accept_invites_only_for_direct_messages = config.get("accept_invites_only_for_direct_messages", False)
        return InviteAutoAccepterConfig(
            accept_invites_only_for_direct_messages=accept_invites_only_for_direct_messages,
        )
```